NanoLog.cpp

```diff
@@ -38,18 +39,22 @@ namespace
     /* Returns microseconds since epoch */
     uint64_t timestamp_now()
     {
-    	return std::chrono::high_resolution_clock::now().time_since_epoch() / std::chrono::microseconds(1);
+    	return std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
     }
 
+    /* I want [2016-10-13 00:01:23.528514] */
     void format_timestamp(std::ostream & os, uint64_t timestamp)
     {
-	uint64_t microseconds = timestamp % 1000000;
-	auto duration = std::chrono::microseconds(timestamp);
-	std::chrono::high_resolution_clock::time_point time_point(duration);
-	std::time_t time_t = std::chrono::high_resolution_clock::to_time_t(time_point);
+	// The next 3 lines do not work on MSVC!
+	// auto duration = std::chrono::microseconds(timestamp);
+	// std::chrono::high_resolution_clock::time_point time_point(duration);
+	// std::time_t time_t = std::chrono::high_resolution_clock::to_time_t(time_point);
+	std::time_t time_t = timestamp / 1000000;
```

Comment:
@Iyengar111 you can take a look at how g3log solves this in a cross platform way. Ref: https://github.com/KjellKod/g3log/blob/master/src/g3log/time.hpp and 

To get the us and ns timestamp you call `g3::timestamp_get(...)`
To get the date/time you call: g3::localtime_formatted(...)  (ref: logmessage.cpp, time.cpp)

It could absolutely be made more cohesive & I will at some point but it works pretty good and it is cross-platform enabled.